extensions/smallrye-graphql-client/runtime/src/main/java/io/quarkus/smallrye/graphql/client/runtime/GraphQLClientConfig.java

```diff
@@ -22,4 +23,17 @@
     @ConfigItem(name = "header")
     public Map<String, String> headers;
 
+    /**
+     * WebSocket subprotocols that should be supported by this client for running subscriptions over websockets.
+     * Allowed values are:
+     * - `graphql-ws` for the deprecated Apollo protocol
+     * - `graphql-transport-ws` for the newer GraphQL over WebSocket protocol
+     * If multiple protocols are provided, the actual protocol to be used will be subject to negotiation with
+     * the server.
+     * To make the client work with the dummy protocol implemented by SmallRye GraphQL 1.4 server-side,
+     * leave this empty.
+     */
+    @ConfigItem(defaultValue = "[]")
+    public List<String> subprotocols;
```

Comment:
Maybe you are aware of it but you can use `Optional<List<String>>` for optional lists. So you could remove the default and I think simplify your code a bit.
Not blocking though and you might have chosen this pattern for a good reason.

Reply:
Ok that's a good point, changed that